buildSrc/subprojects/cleanup/src/main/kotlin/gradlebuild/cleanup.gradle.kts

```diff
@@ -16,49 +16,20 @@
 package gradlebuild
 
 import gradlebuild.basics.BuildEnvironment
-import gradlebuild.cleanup.extension.CleanupExtension
-import gradlebuild.cleanup.services.DaemonTracker
-import gradlebuild.cleanup.tasks.CleanUpCaches
-import gradlebuild.cleanup.tasks.CleanUpDaemons
-import gradlebuild.cleanup.tasks.KillLeakingJavaProcesses
 
 plugins {
     base
-    id("gradlebuild.module-identity") // Would be good to avoid this in the root project, and instead apply a 'clean' Plugin to each subproject
 }
 
-tasks.register<CleanUpCaches>("cleanUpCaches") {
-    version.set(moduleIdentity.version)
-    homeDir.set(layout.projectDirectory.dir("intTestHomeDir"))
-}
-val trackerService = gradle.sharedServices.registerIfAbsent("daemonTracker", DaemonTracker::class) {
-    parameters.gradleHomeDir.fileValue(gradle.gradleHomeDir)
-    parameters.rootProjectDir.set(rootProject.layout.projectDirectory)
-}
-extensions.create<CleanupExtension>("cleanup", trackerService)
-tasks.register<CleanUpDaemons>("cleanUpDaemons") {
-    this.tracker.set(trackerService)
-}
-
-val killExistingProcessesStartedByGradle by tasks.registering(KillLeakingJavaProcesses::class) {
-    tracker.set(trackerService)
-}
-
-// TODO find another solution here to avoid reaching into another project. Maybe the CI job should do 'killExistingProcessesStartedByGradle' first directly.
-if (BuildEnvironment.isCiServer) {
-    tasks {
-        val cleanTask = named("clean") {
-            dependsOn(killExistingProcessesStartedByGradle)
-        }
-        subprojects {
-            this.tasks.configureEach {
-                mustRunAfter(killExistingProcessesStartedByGradle)
+if (BuildEnvironment.isCiServer && rootProject.name != "test") {
+    val killExistingProcessesStartedByGradle by rootProject.tasks.getting
+    val rootCleanTask = rootProject.tasks.clean
 
-                // Workaround for https://github.com/gradle/gradle/issues/2488
-                if (name != "clean") {
-                    mustRunAfter(cleanTask)
-                }
-            }
+    tasks.configureEach {
+        mustRunAfter(killExistingProcessesStartedByGradle)
```

Comment:
If we merge this https://github.com/gradle/gradle/pull/14182 and always clean up after the build,
we should be able to get rid of this plugin altogether

Reply:
The issue is that the whole CI run itself might fail/crash before the step is executed leaving the agent in a bad state. That's why we also want to do the cleanup _before_ we do the actual build.

Reply:
Then I suggest adding an explicit build step that would do this cleanup before running `./gradlew clean *test` and keep the Gradle build clean of `clean` workarounds. I can add this to #14182

Reply:
That sounds like a good idea. Lets do this as part of #14182 and discuss further concerns there.